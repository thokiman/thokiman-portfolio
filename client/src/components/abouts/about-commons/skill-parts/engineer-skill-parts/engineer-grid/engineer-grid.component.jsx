import React from 'react';
import EngineerGridItem from '../engineer-grid-item/engineer-grid-item.component';
import {
  EngineerContentContainer,
  NameEngineerSkill,
} from './enginner-grid.styles';

export const EngineerGrid = ({ engineerSkill }) => {
  return engineerSkill.map(({ id, name, itemsPoint }) => {
    return (
      <EngineerContentContainer key={id}>
        <NameEngineerSkill>{name}</NameEngineerSkill>
        <EngineerGridItem itemsPoint={itemsPoint} />
      </EngineerContentContainer>
    );
  });
};

export default EngineerGrid;
