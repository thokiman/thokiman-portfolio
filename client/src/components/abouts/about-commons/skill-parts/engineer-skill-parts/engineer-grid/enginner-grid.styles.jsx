import styled, { css } from 'styled-components';

const lightBlueColor = '#74c6d4';
const whiteSmokeColor = '#e3e1e4';
const maxWidthSmartphone = 800;
export const EngineerContentContainer = styled.div`
  margin: 4px auto;
  border-top: ${lightBlueColor} solid 2px;
  border-bottom: ${lightBlueColor} solid 2px;
  position: relative;
  top: 10px;
  width: 100%;

  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: repeat (auto, 1fr);
  gap: 0px 5px;
  font-family: baskerville-old-face;
`;
const EngineerSkillStyles = css`
  grid-row: span 3;
  text-align: center;
  font-kerning: auto;
  color: ${whiteSmokeColor};
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const NameEngineerSkill = styled.div`
  ${EngineerSkillStyles}

  padding: 10px;
  font-size: 3vw;
  letter-spacing: 0.04em;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 5vw;
  }
`;

EngineerContentContainer.displayName = 'EngineerContentContainer';
NameEngineerSkill.displayName = 'NameEngineerSkill';
