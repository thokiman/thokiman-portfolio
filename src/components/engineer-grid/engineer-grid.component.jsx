import React from "react";
import { CSSTransitionGroup } from "react-transition-group";

import "./engineer-grid.styles.scss";

const EngineerGrid = ({ engineerSkill }) => {
  return engineerSkill.map(({ id, name, itemsPoint }) => {
    return (
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={600}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <div key={id} className="engineer-content-container">
          <div className="name-engineer-skill">{name}</div>

          {itemsPoint.map(({ id, description }) => {
            return (
              <div key={id} className="description-engineer-skill">
                {description}
              </div>
            );
          })}
        </div>
      </CSSTransitionGroup>
    );
  });
};

export default EngineerGrid;
