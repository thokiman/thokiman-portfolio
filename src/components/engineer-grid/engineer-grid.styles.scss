.engineer-content-container {
    margin: 4px auto;
    border-top: #74c6d4 solid 2px;
    border-bottom: #74c6d4 solid 2px;
    position: relative;
    top: 10px;
    width: 100%;
    height: 50%;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-auto-rows: repeat (auto, 1fr);
    gap: 0px 0px;
    font-family: baskerville-old-face;

    .name-engineer-skill {

        grid-row: span 3;
        padding: 10px;
        text-align: center;
        font-size: 250%;
        font-kerning: auto;
        letter-spacing: 0.04em;
        color: #e3e1e4;
        display: flex;
        align-items: center;
        justify-content: center;

    }



    .description-engineer-skill {

        grid-row: span 3;
        line-height: 1.5em;
        width: 100%;
        text-align: center;
        font-size: 150%;
        font-kerning: auto;
        letter-spacing: 0.03em;
        display: flex;
        align-items: center;
        justify-content: center;
        color: #e3e1e4;

    }


}