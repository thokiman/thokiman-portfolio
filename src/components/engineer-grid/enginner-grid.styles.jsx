import styled, { css } from "styled-components";

export const EngineerContentContainer = styled.div`
  margin: 4px auto;
  border-top: #74c6d4 solid 2px;
  border-bottom: #74c6d4 solid 2px;
  position: relative;
  top: 10px;
  width: 100%;
  height: 50%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: repeat (auto, 1fr);
  gap: 0px 5px;
  font-family: baskerville-old-face;
`;
const EngineerSkillStyles = css`
  grid-row: span 3;
  text-align: center;
  font-kerning: auto;
  color: #e3e1e4;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const NameEngineerSkill = styled.div`
  ${EngineerSkillStyles}

  padding: 10px;
  font-size: 3vw;
  letter-spacing: 0.04em;
`;

EngineerContentContainer.displayName = "EngineerContentContainer";
NameEngineerSkill.displayName = "NameEngineerSkill";
