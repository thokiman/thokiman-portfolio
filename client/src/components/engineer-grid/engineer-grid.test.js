import React from "react";
import { shallow } from "enzyme";

import { EngineerGrid } from "./engineer-grid.component";

describe("EngineerGrid component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      engineerSkill: mockArray,
    };
    wrapper = shallow(<EngineerGrid {...mockProps} />);
  });

  it("should render EngineerGrid component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
