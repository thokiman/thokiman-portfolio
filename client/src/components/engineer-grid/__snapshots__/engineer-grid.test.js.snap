// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineerGrid component should render EngineerGrid component 1`] = `ShallowWrapper {}`;
